src/content/utils.ts

```diff
@@ -36,15 +37,37 @@ export async function extractSection(element: Element) {
 }
 
 export async function findCourseInfo(code: string) {
+  let requestOptions: RequestInit
+  let headers: Headers
+
   const urlencoded = new URLSearchParams()
   urlencoded.append("q", code)
+  urlencoded.append("clientRequestID", generateRandomHexId())
 
-  const requestOptions = {
-    method: "POST",
-    body: urlencoded,
-    redirect: "follow" as RequestRedirect,
-  }
+  if (sessionSecureToken) {
+    urlencoded.append("sessionSecureToken", sessionSecureToken)
 
+    headers = new Headers({
+      "Session-Secure-Token": sessionSecureToken,
+      "Content-Type": "application/x-www-form-urlencoded"
+    })
+
+    requestOptions = {
+      method: "POST",
+      body: urlencoded,
+      redirect: "follow" as RequestRedirect,
+      headers: headers,
+    }
+  } else {
+    requestOptions = {
+      method: "POST",
+      body: urlencoded,
+      redirect: "follow" as RequestRedirect,
+    }
+    headers = new Headers({
+      "Content-Type": "application/x-www-form-urlencoded"
+    })
+  }
```

Comment:
seems like pretty similar "request assembling" functionality has been added to 3 different functions here - is there a way to consolidate this similar code into some kind of helper function?

more broadly, all of these "fetch from workday API in specific way" functions seem pretty similar. we might be able to extract out a whole `fetchFromWorkday` helper function, and just customize URLs/response handling for each use case.

that might be a bit out of scope for this PR though - could leave that for the future.